server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -1707,6 +1701,24 @@ public void trimUnreferencedReaders() throws IOException {
             closeOnTragicEvent(ex);
             throw ex;
         }
+        // Do sync outside the writeLock to avoid blocking all writing thread.
+        try {
```

Comment:
It's easier to follow with `try-catch` inside `if(toDeleteReaderList.isEmpty() == false)`. Here we prefer `if (pred == false)` over `if (!pred)`.